src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp

```diff
@@ -1005,9 +999,6 @@ std::pair<std::time_t, std::size_t> restart_info(const RestartIO::RstState * rst
                     auto injection = std::make_shared<Well::WellInjectionProperties>(well2->getInjectionProperties());
                     injection->handleWCONINJE(record, well2->isAvailableForGroupControl(), well_name);
 
-                    if (well2->updateProducer(false))
```

Comment:
I have absolutely no knowledge of the code, but it seems like the removed method `Well::updateProducer(bool)` might change the status of the well (if the argument is different from the stored one). This step of the algorithm might be lost here.

But probably I am just making noise.

Reply:
Yes you are right that the `Well::updateProducer( )` call changes the status of a well. The call is removed here because it was also called from the `Well:.updateProduction( )` method - i.e. twice.